data/saas/config/google_analytics_config.yml

```diff
@@ -0,0 +1,105 @@
+saas_config:
+  fides_key: <instance_fides_key>
+  name: Google Analytics SaaS Config
+  type: google_analytics
+  description: A schema representing the Google Analytics connector for Fides
+  version: 0.0.1
+
+  connector_params:
+
+    - name: client_id
+    - name: client_secret
+    - name: redirect_uri
+    - name: property_id
+      description: GA4 Property
+
+  client_config:
+    protocol: https
+    host: www.googleapis.com
+    authentication:
+      strategy: oauth2_authorization_code
+      configuration:
+        expires_in: 3599
+        authorization_request:
+          method: GET
+          path: /o/oauth2/v2/auth
+          client_config:
+            protocol: https
+            host: accounts.google.com
+          query_params:
+            - name: client_id
+              value: <client_id>
+            - name: redirect_uri
+              value: <redirect_uri>
+            - name: response_type
+              value: code
+            - name: state
+              value: <state>
+            - name: prompt
+              value: consent
+            - name: access_type
+              value: offline
+            - name: scope
+              value: https://www.googleapis.com/auth/analytics.user.deletion
+        token_request:
+          method: POST
+          path: /token
+          client_config:
+            protocol: https
+            host: oauth2.googleapis.com
+          headers:
+            - name: Content-Type
+              value: application/x-www-form-urlencoded
+          body: |
+            {
+              "client_id": "<client_id>",
+              "client_secret": "<client_secret>",
+              "grant_type": "authorization_code",
+              "code": "<code>",
+              "redirect_uri": "<redirect_uri>"
+            }
+        refresh_request:
+          method: POST
+          client_config:
+            protocol: https
+            host: oauth2.googleapis.com
+          path: /token
+          headers:
+            - name: Content-Type
+              value: application/x-www-form-urlencoded
+          body: |
+            {
+              "client_id": "<client_id>",
+              "client_secret": "<client_secret>",
+              "grant_type": "refresh_token",
+              "refresh_token": "<refresh_token>"
+            }
+
+  test_request:
+    method: POST
+    path: /analytics/v3/userDeletion/userDeletionRequests:upsert
+    body: |
+      {
+        "kind": "analytics#userDeletionRequest",
+        "id": {
+          "type": "USER_ID",
+          "userId": "fides_test_user_id"
+        },
+        "propertyId": "<property_id>"
+      }
+
+  consent_requests:
+    opt_out:
+      method: POST
+      path: /analytics/v3/userDeletion/userDeletionRequests:upsert
+      body: |
+        {
+          "kind": "analytics#userDeletionRequest",
+          "id": {
+            "type": "USER_ID",
+            "userId": "<user_id>"
```

Comment:
this made me pause a bit since in the saas DSR framework we require the identity param reference to be spelled out explicitly with the `param_values.identity` field.

i see [here](https://github.com/ethyca/fides/compare/main...consent-propagation#diff-35fed73a3a47b2a6c6e7d3d413edf8c493406cb51aceb07a28025574fd293c3fR448) that we're allowing for more flexibility/shorthand in the consent request spec. i don't see a huge problem with it, and i'm guessing there's a reason for this - either the functionality is needed or it's meant just as a temporary shortcut to get these types of use cases working well. but could you clarify here why we're doing it this way in the consent request framework?

apologies if you'd explained this somewhere else - i may have just forogtten!

Reply:
oh thanks @adamsachs this is exactly the kind of thing I was hoping to get out of CR!

My intent wasn't to deviate from the SaaS Connector framework, I think I just patterned it off of `SaaSConnector.test_connection` because that populates a single request, and our consent request spec was similar.

But I'd like it to be more similar, not less, I'll look into how to have it use the `param_values.identity` field specifically.

Reply:
OK @adamsachs I created a new `SaaSQueryConfig.generate_consent_stmt` which calls existing `SaaSQueryConfig.generate_update_param_values` and `SaaSQueryConfig.generate_update_request_params`, which enforces the use of `param_values.identity` and `param_values.connector_params` among other things. 

To me, it makes sense to reuse this masking code (rather than access code or creating something brand new) as update/delete requests have a lot of overlap with consent requests (sometimes this involves deleting a user, for example).  Even though we're not using all the attributes in these methods (for example, we're not defining references at the moment between consent collections), we might have to in the future. In the meantime, it forces us to define our consent saas configs similarly to constraints you have in place for access/erasure.

What do you think?
https://github.com/ethyca/fides/pull/2302/commits/45404244a450cff2993e85488af4d6cdb5713958

Reply:
Thanks again @adamsachs for pointing out this discrepancy!